internal/storage/bucket_handle_test.go

```diff
@@ -256,3 +256,51 @@ func (t *BucketHandleTest) TestCreateObjectMethodWhenGivenGenerationObjectNotExi
 	AssertEq(nil, obj)
 	AssertTrue(strings.Contains(err.Error(), "Error 412: Precondition failed"))
 }
+
+func (t *BucketHandleTest) TestListObjectMethodWithValidPrefix() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertNe(nil, obj.Objects)
+	AssertNe(nil, obj.CollapsedRuns)
+	AssertEq(nil, err)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithInValidPrefix() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "InvalidPrefix",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, obj.Objects)
+	AssertEq(nil, obj.CollapsedRuns)
+	AssertEq(nil, err)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithIncludeTrailingDelimiter() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: false,
```

Comment:
also what about tests with includetrailing delimiter, max result and continuation token?

Reply:
fakeserverobject is only checking Prefix,Delimiter.
https://github.com/GoogleCloudPlatform/gcsfuse/blob/9b96e9dc586959661a1ad54622bbb7419e7e3e68/vendor/github.com/fsouza/fake-gcs-server/fakestorage/object.go#L287